ingress-node-firewall/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	// +optional
+	ICMPType uint8 `json:"icmpType,omitempty"`
+
+	// ICMPCode define ICMP Code ID (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=16
+	// +kubebuilder:validation:Minimum:=0
+	// +optional
```

Comment:
Same comments as above, hard to know what this means so perhaps you can make this easier by naming them or providing aliases at least

Also, optional and required?

Reply:
can u explain what u meant by aliases ? it will be a huge list of strings to cover both icmpv4 and icmpv6

Reply:
Yeah I'm suggesting an enum, if possible, the types have a defined set within the spec, do the codes as well?

Reply:
At a minimum, the code and type both seem to have a limit. They are 8 bytes of data at most so, if we keep this numeric, the value must be 0-255 right?

Reply:
I do already have limit for the Type and the code